src/cql/upstream-external/Phq9LogicLibrary.cql

```diff
@@ -114,14 +114,42 @@ define CurrentResponses:
     answer: I.answer.value[0] // Assuming only one answer per question
   }
 
+/*
+ @param question linkId of type FHIR.String
+ @return answer element with that linkId
+ */
+define function getCurrentResponseByQuestionId(qid String):
+  (
+    CurrentResponses R
+    where R.id = qid
+    return R.answer
+  )
+
+/*
+ retrieve concept code of the selected answer option
+ @param question index of type FHIR.Integer
+ @return code value for the answer
+ */
+define function getAnswerOptionCodeValue(qIndex Integer):
+  if Phq9Questionnaire.item[qIndex] is not null then
+    Phq9Questionnaire.item[qIndex].answerOption.value.code.value
+  else
+    null
+
+/*
+ * retrieve the numeric value for the answer from extension
+ * @param a list of elements of type FHIR.Extension
+ * @return the value as Integer
+ */
+define function getAnswerOptionExtensionValue(extList List<FHIR.Extension>):
+  (singleton from (extList e where e.url ~ 'http://hl7.org/fhir/StructureDefinition/ordinalValue')).value.value as Integer
+
 // Question 1
 define Phq9AnswerOptionsQuestion1:
```

Comment:
@achen2401 I really like the direction you're taking in this PR. These duplicative \*Question[n], \*ResponseIndex[n] functions would be nice to generalize too, at some point - I expect you already have that in mind :)